crates/codegen/parser/src/chumsky/boilerplate.rs

```diff
@@ -71,13 +71,18 @@ pub fn cst_head() -> TokenStream {
             Group {
                 children: Vec<Rc<Node>>,
             },
-            // TODO: Error types
         }
 
         impl Node {
             pub fn rule(kind: RuleKind, children: Vec<Option<Rc<Self>>>) -> Option<Rc<Self>> {
                 let children: Vec<_> = children.into_iter().filter_map(|e| e).collect();
-                if children.is_empty() { None } else { Some(Rc::new(Self::Rule { kind, children })) }
+                if children.is_empty() { return None }
+                if children.len() == 1 {
```

Comment:
Nit: I recently learned that descructuring ranges with a specific length is supported:

```rust
if let [Self::Group { children }] = children[..] {
  return Some(Rc::new(Self::Rule { kind, children }));
}
```

I think it would also allow returning the inner `children` directly without cloning.

Reply:
Agreed